repo_refactoring/data/sanity_data/generate_data.py

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on Mon Oct 16 21:43:47 2017
+
+@author: changyaochen
+"""
+
+
+def generate_data(id_filename, save_filename, 
+                  keep_label=True):
+    '''
+    This code is to read the sanity data, from the id
+    
+    Input
+    ---------
+    id_filename: <str>
+        filename for the id file, not including the path
+    save_filename: <str>
+        filename for the 'real' data file
+    keep_label: <Boolean>, default: True
+        If False, remove the label column
+    
+    Return
+    ---------
+    Nothing, but create file(s) under the same folder.
+    If the file(s) already exist(s), we will overwrite
+    '''
+    
+    import pandas as pd
+    import sys
+    # get config files
+    sys.path.append('../../code')
+    import config
+    
+    # get indices
+    with open(id_filename, 'r') as f:
+        idx = f.readlines()
+    idx = list(map(int, idx))
+    
+    # get the data
+    raw_data = '../raw_data/train.csv'
+    raw_df = pd.read_csv(raw_data)
+    df = raw_df[raw_df[config.id_col].isin(idx)]
+    if not keep_label:
```

Comment:
maybe we don't need this keep_label, whether to keep the label should depend on the file type, ie, for train file, it should contain label and id, for test file it should not contain label, and for test_target file, it should only contain id col and label col